smarts/sstudio/types.py

```diff
@@ -462,50 +463,85 @@ class MapZone(Zone):
     """The number of lanes from right to left that this zone covers."""
 
     def to_geometry(self, road_network: SumoRoadNetwork) -> Polygon:
-        def resolve_offset(offset, geometry_length, lane_length, buffer_from_ends):
-            if offset == "base" or offset == 0:
-                return buffer_from_ends
+        def resolve_offset(offset, geometry_length, lane_length):
+            if offset == "base":
+                return 0
             # push off of end of lane
             elif offset == "max":
-                return lane_length - geometry_length - buffer_from_ends
+                return lane_length - geometry_length
             elif offset == "random":
-                return random.uniform(
-                    0, lane_length - geometry_length - buffer_from_ends
-                )
+                return random.uniform(0, lane_length - geometry_length)
             else:
                 return float(offset)
 
+        def pick_remaining_shape_after_split(geometry_collection, length, lane):
+            lane_shape = geometry_collection
+            if not isinstance(lane_shape, GeometryCollection):
+                return lane_shape
+
+            # For simplicty, we only deal w/ the == 1 or 2 case
+            if len(lane_shape) == 0 or len(lane_shape) > 2:
+                return None
+
+            if len(lane_shape) == 1:
+                return lane_shape[0]
+
+            expected_bbox_area = lane.getWidth() * length
+            keep_index = 0
```

Comment:
This could be a loop to check for the right shape out of all shapes.

Reply:
There are only 2.